libs/language-server/src/lib/lsp/jayvee-code-lens-provider.ts

```diff
@@ -0,0 +1,81 @@
+// SPDX-FileCopyrightText: 2024 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+import { type AstNode, type LangiumDocument, type MaybePromise } from 'langium';
+import { type CodeLensProvider } from 'langium/lsp';
+import { type CodeLens } from 'vscode-languageserver';
+import { Command } from 'vscode-languageserver-protocol';
+
+import { isJayveeModel } from '../ast';
+
+export interface RunJayveeCodeLensPayload {
+  filePath: string;
+  pipelineName: string;
+}
+
+export interface ShowPipeOutputCodeLensPayload {
+  pipeOutputName: string;
+  lineNumber: number;
+}
+
+export function isRunJayveeCodeLensPayload(
+  o: unknown,
```

Comment:
Naming could probably be better than `o`, no? At least something like 'payload'.